src/Tracy/Debugger.php

```diff
@@ -470,6 +470,13 @@ public static function timer($name = NULL)
 		$now = microtime(TRUE);
 		$delta = isset($time[$name]) ? $now - $time[$name] : 0;
 		$time[$name] = $now;
+		if (!self::$productionMode) {
+			static $panel;
+			if (!$panel) {
+				$panel = self::getBar()->getPanel('time');
+			}
+			$panel->data[] = array('name' => $name, 'time' => $now, 'delta' => $delta);
+		}
```

Comment:
I would consider doing just 

~~~php
if (!self::$productionMode) {
    self::getBar()->getPanel(__CLASS_ . ':time')->data[] = array('name' => $name, 'time' => $now, 'delta' => $delta);
}
~~~

Your solution suffers from performance asymmetry because first `Debugger::timer()`  call would be slower than the second one.


----

Also only the `$time` is irrelevant. Only `$delta` matter. Therefore you should only do this when `$delta` is not `0`.